searchlib/src/vespa/searchlib/query/query_term_simple.cpp

```diff
@@ -122,94 +160,72 @@ QueryTermSimple::getIntegerRange() const
-QueryTermSimple::getRange() const
+QueryTermSimple::getRange() const noexcept
 {
     return getFloatRange<float>();
 }
 
 template <>
 QueryTermSimple::RangeResult<double>
-QueryTermSimple::getRange() const
+QueryTermSimple::getRange() const noexcept
 {
     return getFloatRange<double>();
 }
 
 template <>
 QueryTermSimple::RangeResult<int8_t>
-QueryTermSimple::getRange() const
+QueryTermSimple::getRange() const noexcept
 {
     return getIntegerRange<int8_t>();
 }
 
 template <>
 QueryTermSimple::RangeResult<int16_t>
-QueryTermSimple::getRange() const
+QueryTermSimple::getRange() const noexcept
 {
     return getIntegerRange<int16_t>();
 }
 
 template <>
 QueryTermSimple::RangeResult<int32_t>
-QueryTermSimple::getRange() const
+QueryTermSimple::getRange() const noexcept
 {
     return getIntegerRange<int32_t>();
 }
 
 template <>
 QueryTermSimple::RangeResult<int64_t>
-QueryTermSimple::getRange() const
+QueryTermSimple::getRange() const noexcept
 {
     return getIntegerRange<int64_t>();
 }
 
-struct IntDecoder {
-    static int64_t fromstr(const char * q, const char * qend, char ** end) noexcept {
-        int64_t v(0);
-        for (;q < qend && isspace(*q); q++);
-        std::from_chars_result err = std::from_chars(q, qend, v, 10);
-        if (err.ec == std::errc::result_out_of_range) {
-            v = (*q == '-') ? std::numeric_limits<int64_t>::min() : std::numeric_limits<int64_t>::max();
-        }
-        *end = const_cast<char *>(err.ptr);
-        return v;
-    }
-    static int64_t nearestDownwd(int64_t n, int64_t min) noexcept { return (n > min ? n - 1 : n); }
-    static int64_t nearestUpward(int64_t n, int64_t max) noexcept { return (n < max ? n + 1 : n); }
-};
-
-struct DoubleDecoder {
-    static double fromstr(const char * q, const char * qend, char ** end) {
-        (void) qend;
-        return vespalib::locale::c::strtod(q, end);
-    }
-    static double nearestDownwd(double n, double min) noexcept { return std::nextafterf(n, min); }
-    static double nearestUpward(double n, double max) noexcept { return std::nextafterf(n, max); }
-};
-
-bool QueryTermSimple::getAsIntegerTerm(int64_t & lower, int64_t & upper) const
+bool QueryTermSimple::getAsIntegerTerm(int64_t & lower, int64_t & upper) const noexcept
 {
     lower = std::numeric_limits<int64_t>::min();
     upper = std::numeric_limits<int64_t>::max();
     return getAsNumericTerm(lower, upper, IntDecoder());
 }
 
-bool QueryTermSimple::getAsDoubleTerm(double & lower, double & upper) const
+bool QueryTermSimple::getAsFloatTerm(double & lower, double & upper) const noexcept
 {
     lower = - std::numeric_limits<double>::max();
     upper =   std::numeric_limits<double>::max();
-    return getAsNumericTerm(lower, upper, DoubleDecoder());
+    return getAsNumericTerm(lower, upper, FloatDecoder<double>());
+}
+
+bool QueryTermSimple::getAsFloatTerm(float & lower, float & upper) const noexcept
+{
+    lower = - std::numeric_limits<float>::max();
```

Comment:
Entirely optional: looks like C++11 has [`std::numeric_limits<T>::lowest()`](https://en.cppreference.com/w/cpp/types/numeric_limits/lowest) which is equal to `- max()` for floating point values, but might be more immediately obvious when reading the code

Reply:
Using lowest.

Reply:
Concluded that +/- infinity is even better.